energymap/client.go

```diff
@@ -0,0 +1,84 @@
+package energymap
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	gridintensity "github.com/thegreenwebfoundation/grid-intensity-go"
+)
+
+type ApiOption func(*ApiClient) error
+
+func New(token string, opts ...ApiOption) (gridintensity.Provider, error) {
+	a := &ApiClient{}
+	for _, opt := range opts {
+		err := opt(a)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if a.client == nil {
+		a.client = &http.Client{
+			Timeout: 5 * time.Second,
+		}
+	}
+
+	if a.apiURL == "" {
+		a.apiURL = "https://api.electricitymap.org/v3"
+	}
+	a.token = token
+
+	if a.moderate_threshold == 0.0 {
+		a.moderate_threshold = 150.0
+	}
+	if a.high_threshold == 0.0 {
+		a.high_threshold = 300.0
+	}
+
+	return a, nil
+}
+
+type ApiClient struct {
+	client             *http.Client
+	apiURL             string
+	token              string
+	moderate_threshold float64
+	high_threshold     float64
```

Comment:
```suggestion
	moderateThreshold float64
	highThreshold     float64
```

Could we use the same casing as the other fields?

Reply:
We don't actually need these as well. This is dangling code from before.